open-apparel-registry/open-apparel-registry
# Filter by List

## Overview

Enables users to filter down to the individual list level. 

Examples for use of this filtering would be the Accord on Fire and Building Safety in Bangladesh consolidating their two separate accounts into one, with users being able to filter to each separate upload; users being able to filter by the different lists for tiers of supply chains uploaded by brands; certification schemes having separate lists for different levels of certification, e.g. Gold, Silver, Bronze, which users can then filter by.

Connects #983 

## Demo

<img width="994" alt="Screen Shot 2021-03-02 at 11 21 10 AM" src="https://user-images.githubusercontent.com/21046714/109681893-eb936080-7b4b-11eb-8da2-7885a9c73502.png">
<img width="642" alt="Screen Shot 2021-03-02 at 11 35 55 AM" src="https://user-images.githubusercontent.com/21046714/109681899-ecc48d80-7b4b-11eb-8342-f9cffc6cb150.png">
<img width="407" alt="Screen Shot 2021-03-02 at 11 36 03 AM" src="https://user-images.githubusercontent.com/21046714/109681908-ee8e5100-7b4b-11eb-8934-1510cc076594.png">
<img width="430" alt="Screen Shot 2021-03-02 at 11 36 10 AM" src="https://user-images.githubusercontent.com/21046714/109681916-efbf7e00-7b4b-11eb-9de3-7fa816b35631.png">

## Testing Instructions

* Run `vagrant ssh` and `./scripts/server`
* Navigate to [/api/docs/#!/contributor-lists/contributor_lists_list](http://localhost:8081/api/docs/#!/contributor-lists/contributor_lists_list)
* Enter 2 as the contributor ID 
    - [x] You should get a list of contributor lists as ID/name pairs
* Navigate to http://localhost:8081/api/docs/#!/facilities/facilities_list
* Set the list's param to 2
    - [x] You should get a list of facilities from the Summer 2019 Compliance List
* Navigate to http://localhost:6543/
    - [x] You should not see the contributor lists
* Select Service Provider A
    - [x] You should see the contributor lists select
* Select the Summer 2019 Compliance List and search
    - [x] You should get a list of facilities from that list
    - [x] The map should load correctly
* Select Civil Society Organization A
    - [x] The previously selected listed should no longer be selected
    - [x] You should see multiple lists in the dropdown
* Select Summer 2019 Apparel List and Summer 2019 Compliance List and search
    - [x] You should get a list of facilities from both lists
* Refresh the page
    - [x] Both contributors and both lists should remain selected
* Navigate to http://localhost:6543/profile/2 
    - [x] The Facility List names should be links
* Click the link
    - [x] The map should open with the appropriate contributor and list selected

## Checklist
